manifests/server/etcd/setup.pp

```diff
@@ -34,6 +67,13 @@
   Optional[Enum['existing', 'new']] $initial_cluster_state = undef,
   Optional[String[1]] $initial_cluster_token               = undef,
   Array[String[1]] $initial_cluster                        = [],
+
+  Stdlib::Unixpath $binary_path  = '/usr/local/bin/etcd',
+  Stdlib::Unixpath $workdir_path = '/var/lib/etcd',
```

Comment:
Calling this `storage_path` would probably make more sense, as it's where the database will be store data on disk. The fact that it happens to be configured by setting the working directory for the service is just an implementation detail.

Keep the name `workdir_path` for the service unit template though, it makes plenty of sense there.